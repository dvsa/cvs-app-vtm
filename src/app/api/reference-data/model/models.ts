export * from './reference-data.model';
