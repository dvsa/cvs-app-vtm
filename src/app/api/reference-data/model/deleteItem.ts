/**
 * Reference Data Microservice
 * This is the API spec for the reference data service. The reference data will be stored in a AWS DynamoDB database.
 *
 * OpenAPI spec version: 1.0.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */

export interface DeleteItem { 
    success?: boolean;
}