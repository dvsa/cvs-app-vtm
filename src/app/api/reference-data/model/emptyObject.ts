/**
 * Reference Data Microservice
 * This is the API spec for the reference data service. The reference data will be stored in a AWS DynamoDB database.
 *
 * OpenAPI spec version: 1.0.0
 * Contact: test@test.com
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface EmptyObject {
}
