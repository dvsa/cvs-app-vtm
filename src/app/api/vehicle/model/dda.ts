/**
 * Disability Discrimination Act
 */
export interface Dda { 
    /**
     * Used only for PSV
     */
    certificateIssued?: boolean;
    /**
     * Used only for PSV
     */
    wheelchairCapacity?: number;
    /**
     * Used only for PSV
     */
    wheelchairFittings?: string;
    /**
     * Used only for PSV
     */
    wheelchairLiftPresent?: boolean;
    /**
     * Used only for PSV
     */
    wheelchairLiftInformation?: string;
    /**
     * Used only for PSV
     */
    wheelchairRampPresent?: boolean;
    /**
     * Used only for PSV
     */
    wheelchairRampInformation?: string;
    /**
     * Used only for PSV
     */
    minEmergencyExits?: number;
    /**
     * Used only for PSV
     */
    outswing?: string;
    /**
     * Used only for PSV
     */
    ddaSchedules?: string;
    /**
     * Used only for PSV
     */
    seatbeltsFitted?: number;
    /**
     * Used only for PSV
     */
    ddaNotes?: string;
}