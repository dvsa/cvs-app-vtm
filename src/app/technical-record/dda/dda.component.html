<dl class="govuk-summary-list">
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">DDA certificate issued (Yes/No)</dt>
    <dd id="dda-certificateIssued" class="govuk-summary-list__value">
      {{ ddaDetails?.certificateIssued | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>

  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">Wheelchair capacity</dt>
    <dd id="dda-wheelchairCapacity" class="govuk-summary-list__value">
      {{ ddaDetails?.wheelchairCapacity | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>

  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">Wheelchair fittings</dt>
    <dd id="dda-wheelchairFittings" class="govuk-summary-list__value">
      {{ ddaDetails?.wheelchairFittings | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>

  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">Wheelchair lift present</dt>
    <dd id="dda-wheelchairLiftPresent" class="govuk-summary-list__value">
      {{ ddaDetails?.wheelchairLiftPresent | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>

  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">Wheelchair lift information</dt>
    <dd id="dda-wheelchairLiftInformation" class="govuk-summary-list__value">
      {{ ddaDetails?.wheelchairLiftInformation | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>

  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">Wheelchair ramp present</dt>
    <dd id="dda-wheelchairRampInformation" class="govuk-summary-list__value">
      {{ ddaDetails?.wheelchairRampInformation | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>

  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">Wheelchair ramp information</dt>
    <dd id="dda-wheelchairRampInformation" class="govuk-summary-list__value">
      {{ ddaDetails?.wheelchairRampInformation | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>

  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">Minimum emergency exits needed</dt>
    <dd id="dda-minEmergencyExits" class="govuk-summary-list__value">
      {{ ddaDetails?.minEmergencyExits | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>

  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">Outswing</dt>
    <dd id="dda-outswing" class="govuk-summary-list__value">
      {{ ddaDetails?.outswing | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>

  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">DDA schedules</dt>
    <dd id="dda-ddaSchedules" class="govuk-summary-list__value">
      {{ ddaDetails?.ddaSchedules | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>

  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">Seatbelts fitted</dt>
    <dd id="dda-seatbeltsFitted" class="govuk-summary-list__value">
      {{ ddaDetails?.seatbeltsFitted | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>

  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">DDA notes</dt>
    <dd id="dda-ddaNotes" class="govuk-summary-list__value">
      {{ ddaDetails?.ddaNotes | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>

</dl>
