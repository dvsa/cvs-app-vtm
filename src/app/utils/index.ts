export * from './mockStore';
export * from './test-model.utils';
export * from './testing.utils';
export * from './form.utils';


