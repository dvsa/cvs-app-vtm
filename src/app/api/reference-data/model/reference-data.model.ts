import { ReferenceDataResourceType } from '@models/reference-data.model';

export type ReferenceDataItem = {
  resourceType: ReferenceDataResourceType;
  resourceKey: string | number;
};

export type ReferenceDataList = Array<ReferenceDataItem>;

export type ReferenceDataApiResponse = {
  data: ReferenceDataList;
};
