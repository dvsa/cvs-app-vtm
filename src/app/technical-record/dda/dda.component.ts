import { Component, OnInit, ChangeDetectionStrategy, Input } from '@angular/core';
import { TechRecordHelpersService } from '@app/technical-record/tech-record-helpers.service';
import { DDA } from '@app/models/tech-record.model';

@Component({
  selector: 'vtm-dda',
  templateUrl: './dda.component.html',
  styleUrls: ['../../app.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class DdaComponent implements OnInit {

  @Input() ddaDetails: DDA;

  constructor(public techRecHelpers: TechRecordHelpersService) { }

  ngOnInit() {}
}
