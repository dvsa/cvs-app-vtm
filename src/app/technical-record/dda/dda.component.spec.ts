import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { DdaComponent } from './dda.component';
import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { SharedModule } from '@app/shared/shared.module';
import { TESTING_UTILS } from '@app/utils/testing.utils';
import { DDA } from '@app/models/tech-record.model';

describe('DdaComponent', () => {
  let component: DdaComponent;
  let fixture: ComponentFixture<DdaComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [SharedModule],
      declarations: [DdaComponent]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(DdaComponent);
    component = fixture.componentInstance;
  });
  
  it('should create view only with populated data', () => {
    component.ddaDetails = TESTING_UTILS.mockDDA();
    fixture.detectChanges();

    expect(component).toBeDefined();
    expect(fixture).toMatchSnapshot();
  });
});
