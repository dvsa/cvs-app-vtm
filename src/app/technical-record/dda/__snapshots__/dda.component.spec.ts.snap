// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DdaComponent should create view only with populated data 1`] = `
<vtm-dda
  ddaDetails={[Function Object]}
  techRecHelpers={[Function TechRecordHelpersService]}
>
  <dl
    class="govuk-summary-list"
  >
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        DDA certificate issued (Yes/No)
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="dda-certificateIssued"
      >
         Yes 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        Wheelchair capacity
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="dda-wheelchairCapacity"
      >
         23 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        Wheelchair fittings
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="dda-wheelchairFittings"
      >
         23 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        Wheelchair lift present
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="dda-wheelchairLiftPresent"
      >
         none 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        Wheelchair lift information
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="dda-wheelchairLiftInformation"
      >
         none 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        Wheelchair ramp present
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="dda-wheelchairRampInformation"
      >
         none 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        Wheelchair ramp information
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="dda-wheelchairRampInformation"
      >
         none 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        Minimum emergency exits needed
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="dda-minEmergencyExits"
      >
         3 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        Outswing
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="dda-outswing"
      >
         no 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        DDA schedules
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="dda-ddaSchedules"
      >
         123 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        Seatbelts fitted
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="dda-seatbeltsFitted"
      >
         3 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        DDA notes
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="dda-ddaNotes"
      >
         all 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
  </dl>
</vtm-dda>
`;
